tuning/sft_trainer.py

```diff
@@ -29,6 +31,40 @@ def on_save(self, args, state, control, **kwargs):
             os.remove(os.path.join(checkpoint_path, "pytorch_model.bin"))
 
 
+class FileLoggingCallback(TrainerCallback):
+    """Exports metrics, e.g., training loss to a file in the checkpoint directory."""
+    # Keys to be exported from the log dict
+    log_keys = ["loss", "epoch"]
+
+    def __init__(self, logger):
+        self.logger = logger
+        self.existing_logs = []
+
+    def on_log(self, args, state, control, logs=None, **kwargs):
+        """Checks if this log contains keys of interest, e.g., los, and if so, creates
+        train_loss.json in the model output dir (if it doesn't already exist),
+        appends the subdict of the log & dumps the file.
+        """
+
+        log_file_path = os.path.join(args.output_dir, "train_loss.json")
```

Comment:
Can we add either a conversion or a check to make sure `loss` is actually in `float` instead of other data types. Otherwise the json dump would fail.